File: trail2/개발팀의 능력/the-capabilities-of-the-development-team.java
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int[] arr = new int[5];

        int ans = Integer.MAX_VALUE;
        boolean f = false;

        for (int i = 0; i < 5; i++) {
            arr[i] = sc.nextInt();
        }
        // Please write your code here.
        
        int total = 0;

        for (int i = 0; i < 5; i++) {
            total += arr[i];
        }

        for (int i = 0; i < 5; i++) {
            int sum1 = arr[i];

            for (int j = 0; j < 5; j++) {
                if (i == j) {
                    continue;
                }

                for (int k = 0; k < 5; k++) {
                    if (i == k || j == k) {
                        continue;
                    }

                    int sum2 = arr[j] + arr[k];
                    int sum3 = total - (sum1 + sum2);

                    if (sum1 == sum2 || sum2 == sum3 || sum3 == sum1) {
                        f = false;
                        break;
                    }
                    ans = Math.min(ans, Math.max(sum1, Math.max(sum2, sum3)) - Math.min(sum1, Math.min(sum2, sum3)));
                    f = true;
                }
            }
        }

        if (f) {
            System.out.println(ans);
        }
        else {
            System.out.println(-1);
        }
    }
}

File: trail2/개발팀의 능력/the-capabilities-of-the-development-team.py
import sys

arr = list(map(int, input().split()))

# Please write your code here.
n = len(arr)

ans = sys.maxsize

total = sum(arr)

f = True

for i in range(n):
        sum1 = arr[i]
        
        for j in range(n):
            if j == i:
                continue
            
            for k in range(n):
                if k == i or k == j:
                    continue

                sum2 = arr[j] + arr[k]
                sum3 = total - (sum1 + sum2)
                
                if sum1 == sum2 or sum2 == sum3 or sum3 == sum1:
                    f = False
                    continue
                f= True
                ans = min(ans, max(sum1, sum2, sum3) - min(sum1, sum2, sum3))

if f:
    print(ans)
else:
    print(-1)